packages/@aws-cdk/cloudformation-diff/lib/iam/statement.ts

```diff
@@ -32,36 +32,52 @@ export class Statement {
    */
   public readonly condition?: any;
 
-  constructor(statement: UnknownMap) {
-    this.sid = expectString(statement.Sid);
-    this.effect = expectEffect(statement.Effect);
-    this.resources = new Targets(statement, 'Resource', 'NotResource');
-    this.actions = new Targets(statement, 'Action', 'NotAction');
-    this.principals = new Targets(statement, 'Principal', 'NotPrincipal');
-    this.condition = statement.Condition;
+  private readonly serializedIntrinsic: string | undefined;
+
+  constructor(statement: UnknownMap | string) {
+    if (typeof statement === 'string') {
```

Comment:
I don't get what this object is invoked with if it's not an object.

A JSON-ified object? But that's not a change, was that already there? Because that seems very silly.